Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -399,18 +261,31 @@ private extension UInt8 {
 
 extension HTTPHeaders {
     func isKeepAlive(version: HTTPVersion) -> Bool {
-        switch self._storage.keepAliveState {
+        switch self.keepAliveState {
         case .close:
             return false
         case .keepAlive:
             return true
         case .unknown:
-            switch self.keepAliveFromHeaders {
-            case .keepAlive:
-                return true
+            var state = KeepAliveState.unknown
+            for value in self["connection"] {
+                for word in value.split(separator: ",").makeIterator() {
```

Comment:
This code should use the `canonicalForm:` subscript. Alternatively, if you are avoiding that for performance reasons, the call to `makeIterator` is unnecessary here.